src/box/lua/trigger.c

```diff
@@ -11,6 +11,67 @@
 #include "lua/utils.h"
 #include "tt_static.h"
 
+#include "box/port.h"
+
+#include "core/func_adapter.h"
+
+/**
+ * A wrapper over non-Lua trigger handlers.
+ * Must be pushed with 2 upvalues:
+ * 1. Event name
+ * 2. Trigger name
+ */
+static int
+luaT_trigger_non_lua_handler(struct lua_State *L)
+{
+	const char *event_name = lua_tostring(L, lua_upvalueindex(1));
+	const char *trigger_name = lua_tostring(L, lua_upvalueindex(2));
+
+	struct func_adapter *trigger = NULL;
+	struct event *event = event_get(event_name, false);
+	if (event != NULL)
+		trigger = event_find_trigger(event, trigger_name);
+	if (trigger == NULL)
+		luaL_error(L, "Trigger was deleted");
+
+	struct port args, ret;
+	port_lua_create(&args, L);
+	bool fiber_storage_is_used = fiber()->storage.lua.stack == L;
+	/*
+	 * Prevent using the same Lua-stack in trigger because port_lua
+	 * doesn't allow to dump it to the same stack as the underlying one.
+	 */
+	if (fiber_storage_is_used)
+		fiber()->storage.lua.stack = NULL;
+	int rc = func_adapter_call(trigger, &args, &ret);
```

Comment:
Looks like code duplication with `luaT_trigger_call`.

Reply:
After rework these function are not similar.